tests/test_notebooks.py

```diff
@@ -10,49 +10,49 @@ def test_notebook_service_available(notebook_service):
     assert response.status_code == 200
 
 
-def test_process_list(selenium_driver, screenshot_dir):
+def test_process_list(selenium_driver, screenshot):
     driver = selenium_driver("notebooks/process_list.ipynb")
+    screenshot["name"] = "process-list.png"
     driver.find_element(By.XPATH, '//button[text()="Update now"]')
-    driver.get_screenshot_as_file(f"{screenshot_dir}/process-list.png")
 
 
-def test_aiida_datatypes_viewers(selenium_driver, screenshot_dir):
+def test_aiida_datatypes_viewers(selenium_driver, screenshot):
     driver = selenium_driver("notebooks/aiida_datatypes_viewers.ipynb")
+    screenshot["name"] = "datatypes-viewer.png"
     driver.set_window_size(1000, 2000)
     driver.find_element(By.CLASS_NAME, "widget-label")
     driver.find_element(By.XPATH, '//button[text()="Clear selection"]')
     time.sleep(5)
-    driver.get_screenshot_as_file(f"{screenshot_dir}/datatypes-viewer.png")
 
 
-def test_eln_configure(selenium_driver, screenshot_dir):
+def test_eln_configure(selenium_driver, screenshot):
     driver = selenium_driver("notebooks/eln_configure.ipynb")
+    screenshot["name"] = "eln-configure.png"
     driver.find_element(By.XPATH, '//button[text()="Set as default"]')
-    driver.get_screenshot_as_file(f"{screenshot_dir}/eln-configure.png")
 
 
-def test_process(selenium_driver, screenshot_dir):
+def test_process(selenium_driver, screenshot):
     driver = selenium_driver("notebooks/process.ipynb")
+    screenshot["name"] = "process.png"
```

Comment:
Is the fixture must be called at the very beginning or it can be moved to the last which makes more sense?

Reply:
I think running this at the beginning is better. This line sets the screenshot filename and puts on hold the rest of the code that follows `yield`.

The code after `yield` will be executed once all the tests are completed. More information can be found following the [link](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended) shared by @danielhollas.